src/Sarif/Core/WebRequest.cs

```diff
@@ -99,35 +106,38 @@ private static Uri SynthesizeAbsoluteUriFrom(Uri uri)
         const string QueryPattern =
             @"^
               \?                              # A literal '?', followed by zero or more
-              (
-                (?<name>[^=])                 # parameter name (everything that's not an = sign),
-                =                             # an equal sign, and
+              (?<query>
                 (
-                  (?<value>[^&]*)             # the value (everything that's not an '&')...
-                  &?                          # and an '&' (except after the last one).
-                )
-              )*";
+                  (?<name>[^=]+)                # parameter name (everything that's not an = sign),
+                  =                             # an equal sign, and
+                  (
+                    (?<value>[^&]*)             # the value (everything that's not an '&')...
+                    &?                          # and an '&' (except after the last one).
+                  )
+                )*
+                |                               # RFC 3986 does _not_ require a URI's query to
```

Comment:
>RFC 3986 does _not_ require a [](start = 50, length = 29)

thanks for this comment, seeing this made me go read a bit reg rfc3986, and that made me think, we are currently not handling the possible presence of #<fragment> post the query. do we care?

Reply:
Here's the definition for quick reference - i couldnt judge if its a useful parameter for us to store in sarif: The fragment identifier component of a URI allows indirect identification of a secondary resource by reference to a primary resource and additional identifying information.

---
In reply to: [296963117](https://github.com/microsoft/sarif-sdk/pull/1545#discussion_r296963117) [](ancestors = 296963117)

Reply:
That's a good question. The short answer is that RFC 7230 (HTTP) and RFC 3986 (URI) together prohibit a fragment here. The long answer is fun :-) :

RFC 7230 §3.1.1 says that a request message begins with a `request-line`:
```
request-line = method SP request-target SP HTTP-version CRLF
```
For example:
```
GET /hello.txt HTTP/1.1
```
We are concerned with the part in the middle, the `request-target`.

RFC 7230 §5.3 tells us that `request-target` can take one of four forms: `origin-form`, `absolute-form`, `authority-form`, and `asterisk-form`.

§5.3.1 says that `origin-form` is `absolute-path [ "?" query ]`. So, no fragment.

§5.3.2 says that `absolute-form` is `absolute-URI`, which we have to look up in RFC 3986, where we learn that it's `scheme ":" hier-part [ "?" query ]`. Again, no fragment.

§5.3.3 says that `authority-form` is just `authority`, which again we have to look up in RFC 3986, where we learn that it's `[ userinfo "@" ] host [ ":" port ]`. In full generality, it might look like `harleen@www.example.com:7700`. Usually it's just `www.example.com`.

§5.3.4 says that `asterisk-form` (§5.3.4) is just an asterisk: `*`.

---
In reply to: [296964466](https://github.com/microsoft/sarif-sdk/pull/1545#discussion_r296964466) [](ancestors = 296964466,296963117)

Reply:
wow, thanks for the detailed explanation!

---
In reply to: [297001721](https://github.com/microsoft/sarif-sdk/pull/1545#discussion_r297001721) [](ancestors = 297001721,296964466,296963117)